src/nodes.coffee

```diff
@@ -1666,20 +1784,30 @@ exports.Code = class Code extends Base
       node.error "multiple parameters named #{name}" if name in uniqs
       uniqs.push name
     @body.makeReturn() unless wasEmpty or @noReturn
-    code = 'function'
-    code += '*' if @isGenerator
-    code += ' ' + @name if @ctor
-    code += '('
-    answer = [@makeCode(code)]
+    code = []
+    code.push @makeCode 'static ' if @static
+    code.push @makeCode 'function' unless @isMethod
+    code.push @makeCode '*' if @isGenerator
+    if @isMethod
+      name = @name.compileToFragments o
+      name.shift() if name[0].code is '.'
+      code.push name...
+
+      if @isDerivedCtor then for param in @params when param.name.this
```

Comment:
Derived constructors must call `super` before using `this`, which conflicts with `@` arguments going at the top of the constructor.

I think the best way to deal with this would be to automatically call `super` as the first statement in all derived constructors, disallowing it from ever being written explicitly. This would have implications with changing the arguments passed to superclasses and writing factory constructors that want to `return` other instances.